### PR TITLE
[CM-1180] Added Support for App-Extension

### PR DESCRIPTION
**Pull request by Sathyan-KM**
## Summary
- Added Support for App-extensions
- calling `UIApplication.shared` directly is not supported in App extensions. Added UIApplication+Extension file

### CODE REVIEW

File: Sources/Utilities/UIApplication+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+//  UIApplication+Extension.swift
+//  KommunicateChatUI-iOS-SDK
+//
+//  Created by sathyan elangovan on 07/02/23.
+//
+
+import Foundation
+
+public extension UIApplication {
+    static var main: UIApplication? { UIApplication.value(forKeyPath: #keyPath(UIApplication.shared)) as? UIApplication }
```
Comment (mukesh-komm) [inline Sources/Utilities/UIApplication+Extension.swift:11]: Won't it still throw a warning if we are using `UIApplication.shared`? We can mark the method as unavailable:
```
@available(iOSApplicationExtension, unavailable, message: "UIApplication.shared is unavailable in application extensions")
```

Comment (Sathyan-KM) [reply Sources/Utilities/UIApplication+Extension.swift:11]: if we make this unavailable to extensions, they can't use this in app-extension target right ? Also they are using this same approach in their app. @mukesh-komm

Comment (mukesh-komm) [reply Sources/Utilities/UIApplication+Extension.swift:11]: This method won't be directly used from extensions, so it should not throw any error. I am not sure, check once in sample app if possible.

Comment (Sathyan-KM) [reply Sources/Utilities/UIApplication+Extension.swift]: changed to function & made it unavailable for app-extension

**Review by adarshmishra — APPROVED** [review]

**Pull request merged**